python_modules/dagster/dagster/_core/execution/context/input.py

```diff
@@ -219,21 +197,31 @@ def op_def(self) -> "OpDefinition":
     @public  # type: ignore
     @property
     def config(self) -> Any:
+        """The config attached to the input that we’re loading."""
         return self._config
 
     @public  # type: ignore
     @property
     def metadata(self) -> Optional[Mapping[str, Any]]:
+        """A dict of metadata that is assigned to the InputDefinition that we’re loading for. 
+        This property only contains metadata passed in explicitly with AssetIn or In. 
+        To access metadata of an upstream asset or operation definition, 
+        use the metadata in InputContext.upstream_output."""
         return self._metadata
 
     @public  # type: ignore
     @property
     def upstream_output(self) -> Optional["OutputContext"]:
+        """Info about the output that produced the object we’re loading."""
         return self._upstream_output
 
     @public  # type: ignore
     @property
     def dagster_type(self) -> "DagsterType":
+        """The type of this input. Dagster types do not propagate from an upstream output 
```

Comment:
I would say "The Dagster type of this input", because Dagster types can be different than Python types, and it's good to be specific about which one we're talking about.